spec/lib/utils/job-utils/http-tool-spec.js

```diff
@@ -0,0 +1,178 @@
+// Copyright, 2016, EMC, Inc.
+/* jshint node: true */
+
+'use strict';
+var nock = require('nock');
+
+describe("HttpTool", function(){
+    var httpTool;
+    var siteGen = 'http://mysite.emc.com';
+    var requestSettings = {};
+    
+
+    before(function(){
+        helper.setupInjector(
+            _.flattenDeep([
+                helper.require('/lib/utils/job-utils/http-tool.js')
+            ])
+        );
+        var HttpTool = helper.injector.get('JobUtils.HttpTool');
+        httpTool = new HttpTool();
+    });
+
+    it('Should handle basic auth', function(done){
+        nock(siteGen).get('/basicAuth')
+        .basicAuth({user:'hello', pass:'world'})
+        .reply(200);
+
+        requestSettings.url = siteGen + '/basicAuth';
+        requestSettings.method = 'GET';
+        requestSettings.credential = {username:'hello', password:'world'};
+        // requestSettings.data = '{hello: world}';
+
+        httpTool.setupRequest(requestSettings)
+        .then(httpTool.runRequest()
```

Comment:
if you want your style code, don't call the function, so the following is correct:
```javascript
.then(httpTool.runRequest)
```